test-framework/common/src/main/java/io/quarkus/test/common/QuarkusTestResource.java

```diff
@@ -23,6 +23,13 @@
      */
     Class<? extends QuarkusTestResourceLifecycleManager> value();
 
+    /**
+     * @return The arguments to be passed to the {@code QuarkusTestResourceLifecycleManager}
+     *
+     * @see QuarkusTestResourceLifecycleManager#init(String[])
+     */
+    String[] initArgs() default {};
```

Comment:
Wouldn't it make more sense for this to be a key-value type arrangement? You can do this by creating a new @ResourceArg annotation with name+value and using an array of annotations.

Reply:
Yes, that sounds like a good idea. That way only the values need to be deserialized (if not used as strings). The signature of `init()` could change to `init(Map<String, String> args)`.

Reply:
That's an excellent idea @stuartwdouglas!  I updated the PR to include it.